mailpoet/lib/Mailer/Methods/SMTP.php

```diff
@@ -53,89 +45,25 @@ public function __construct(
     $this->login = $login;
     $this->password = $password;
     $this->encryption = $encryption;
-    $this->sender = $sender;
-    $this->replyTo = $replyTo;
-    $this->returnPath = $returnPath;
-    $this->mailer = $this->buildMailer();
-    $this->mailerLogger = new Swift_Plugins_Loggers_ArrayLogger();
-    $this->mailer->registerPlugin(new Swift_Plugins_LoggerPlugin($this->mailerLogger));
-    $this->errorMapper = $errorMapper;
-    $this->blacklist = new BlacklistCheck();
+    parent::__construct($sender, $replyTo, $returnPath, $errorMapper);
   }
 
-  public function send($newsletter, $subscriber, $extraParams = []): array {
-    if ($this->blacklist->isBlacklisted($subscriber)) {
-      $error = $this->errorMapper->getBlacklistError($subscriber);
-      return Mailer::formatMailerErrorResult($error);
-    }
-    try {
-      $message = $this->createMessage($newsletter, $subscriber, $extraParams);
-      $result = $this->mailer->send($message);
-    } catch (\Exception $e) {
-      return Mailer::formatMailerErrorResult(
-        $this->errorMapper->getErrorFromException($e, $subscriber)
-      );
-    }
-    if ($result === 1) {
-      return Mailer::formatMailerSendSuccessResult();
-    } else {
-      $error = $this->errorMapper->getErrorFromLog($this->mailerLogger->dump(), $subscriber);
-      return Mailer::formatMailerErrorResult($error);
-    }
-  }
+  public function buildMailer(): PHPMailer {
+    $mailer = new PHPMailer(true);
+    $mailer->isSMTP();
+    $mailer->Host = $this->host; // phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+    $mailer->Port = $this->port; // phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+    $mailer->SMTPSecure = $this->encryption;
+    /** @phpstan-ignore-next-line - we cannot annotate the return type from a filter */
+    $mailer->SMTPOptions = $this->wp->applyFilters('mailpoet_mailer_smtp_options', []);
+    /** @phpstan-ignore-next-line - we cannot annotate the return type from a filter */
+    $mailer->Timeout = $this->wp->applyFilters('mailpoet_mailer_smtp_connection_timeout', self::SMTP_CONNECTION_TIMEOUT); // phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
```

Comment:
Can't you do like `$mailer->Timeout = (int) $this->wp->applyFilters()`? Or is the problem something else?

Reply:
I tried it, but it doesn't work. PHPstan reports an error `Cannot cast mixed to int.`
It makes sense to me because the filter can return anything.